### ISSUE TITLE
Add more warnings to Jenkins

### ISSUE THREAD

**Issue by aprokop** [ORNL-CEES/DataTransferKit#130]
I think to we need to further clean up the code to get rid of warnings. Thus I suggest we add -Werror and additional warnings detection to Jenkins CI. At the moment I get things like this:

```
[1059/1294] Building CXX object DataTransferKit/packages/Operators/test/CMakeFiles/dtk_hex_test_reference.dir/reference_implementation/DTK_ReferenceHexMesh.cpp.o
In file included from ../../DataTransferKit/packages/Interface/src/Client/DTK_EntitySet.hpp:47:0,
                 from ../../DataTransferKit/packages/Interface/src/OperatorVector/DTK_FunctionSpace.hpp:46,
                 from ../../DataTransferKit/packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.hpp:45,
                 from ../../DataTransferKit/packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp:41:
../../DataTransferKit/packages/Interface/src/Client/DTK_EntityIterator.hpp:91:28: warning: ‘virtual DataTransferKit::EntityIterator DataTransferKit::EntityIterator::operator++(int)’ was hidden [-Woverloaded-virtu
al]
     virtual EntityIterator operator++( int );
                            ^~~~~~~~
In file included from ../../DataTransferKit/packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp:50:0:
../../DataTransferKit/packages/Adapters/BasicGeometry/src/DTK_BasicEntitySet.hpp:83:21: warning:   by ‘virtual DataTransferKit::EntityIterator& DataTransferKit::BasicEntitySetIterator::operator++()’ [-Woverloaded
-virtual]
     EntityIterator &operator++() override;
                     ^~~~~~~~
```

One thing, though, is that some of Trilinos packages like STK are more warning-prone. I think the linear algebra portion, though, is mostly warnings free.

**Comment by sslattery** [comment]
Remember that we get warnings from TPLs when we build too that we have no control over. Would this flag cause the build to fail because a warning from a TPL?

**Comment by aprokop** [comment]
I usually have the following in my configure script:
```bash
SYSTEM_HEADERS=""
for header_dir in `echo $CPATH | sed 's/:/\ /g'`; do
  SYSTEM_HEADERS="$SYSTEM_HEADERS -isystem $header_dir"
done

...

-D CMAKE_CXX_FLAGS="-g -Wall -Wextra -Wno-unused-parameter $CXX_FLAGS_COMPILER_SPECIFIC $SYSTEM_HEADERS"
```
This way the compiler would ignore the headers coming from the things in you $CPATH (or whatever you put there). So TPLs should be fine. Most issues I see coming from Trilinos packages.

**Comment by sslattery** [comment]
I think this is a good idea then as long as we can configure this to just fail on any DTK warnings.

**Comment by dalg24** [comment]
The problem is not so much the TPL but the other packages in Trilinos.

**Comment by aprokop** [comment]
Fixed in #134.
